test/e2e/storage/persistent_volumes-local.go

```diff
@@ -252,6 +253,86 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
 		})
 	})
 
+	Context("when pod using local volume with non-existant path", func() {
+		ep := &eventPatterns{
+			reason:  "FailedMount",
+			pattern: make([]string, 2)}
+		ep.pattern = append(ep.pattern, "MountVolume.SetUp failed")
+		ep.pattern = append(ep.pattern, "does not exist")
+
+		It("should not be able to mount", func() {
+			for _, testVolType := range LocalVolumeTypes {
+				By(fmt.Sprintf("local-volume-type: %s", testVolType))
+				testVol := &localTestVolume{
+					node:            config.node0,
+					hostDir:         "/non-existent/location/nowhere",
+					localVolumeType: testVolType,
+				}
+				By("Creating local PVC and PV")
+				createLocalPVCPV(config, testVol)
+				pod, err := createLocalPod(config, testVol)
+				Expect(err).To(HaveOccurred())
+				checkPodEvents(config, pod.Name, ep)
+			}
+		})
+	})
+
+	Context("when pod's node is different from PV's NodeAffinity", func() {
+
+		BeforeEach(func() {
+			if len(config.nodes.Items) < 2 {
+				framework.Skipf("Runs only when number of nodes >= 2")
+			}
+		})
+
+		ep := &eventPatterns{
+			reason:  "FailedScheduling",
+			pattern: make([]string, 2)}
+		ep.pattern = append(ep.pattern, "MatchNodeSelector")
+		ep.pattern = append(ep.pattern, "No nodes are available")
+		for _, testVolType := range LocalVolumeTypes {
+
+			It("should not be able to mount due to different NodeAffinity", func() {
+				var testVol *localTestVolume
+				By(fmt.Sprintf("local-volume-type: %s", testVolType))
+				testVol = setupLocalVolumePVCPV(config, testVolType)
+				pod := makeLocalPodWithNodeAffinity(config, testVol, config.nodes.Items[1].Name)
```

Comment:
All the steps except makeLocalPod are the same across all three tests.  Can you refactor into a common function?